libs/community/tests/integration_tests/document_loaders/test_sql_database.py

```diff
@@ -0,0 +1,233 @@
+"""
+Test SQLAlchemy document loader functionality on behalf of SQLite and PostgreSQL.
+"""
+import functools
+import logging
+import typing
+import warnings
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+import pytest
+import sqlalchemy as sa
+import sqlparse
+
+from langchain_community.utilities.sql_database import SQLDatabase
+
+if typing.TYPE_CHECKING:
+    from _pytest.python import Metafunc
+
+from langchain_community.document_loaders.sql_database import SQLDatabaseLoader
+from tests.data import MLB_TEAMS_2012_SQL
+
+logging.basicConfig(level=logging.DEBUG)
+
+
+try:
+    import sqlite3  # noqa: F401
+
+    sqlite3_installed = True
+except ImportError:
+    warnings.warn("sqlite3 not installed, skipping corresponding tests", UserWarning)
+    sqlite3_installed = False
+
+try:
+    import psycopg2  # noqa: F401
+
+    psycopg2_installed = True
+except ImportError:
+    warnings.warn("psycopg2 not installed, skipping corresponding tests", UserWarning)
+    psycopg2_installed = False
+
+
+@pytest.fixture()
+def engine(db_uri: str) -> sa.Engine:
+    """
+    Return an SQLAlchemy engine object.
+    """
+    return sa.create_engine(db_uri, echo=False)
+
+
+@pytest.fixture()
+def db(engine: sa.Engine) -> SQLDatabase:
+    return SQLDatabase(engine=engine)
+
+
+@pytest.fixture()
+def provision_database(engine: sa.Engine) -> None:
+    """
+    Provision database with table schema and data.
+    """
+    sql_statements = MLB_TEAMS_2012_SQL.read_text()
+    with engine.connect() as connection:
+        connection.execute(sa.text("DROP TABLE IF EXISTS mlb_teams_2012;"))
+        for statement in sqlparse.split(sql_statements):
+            connection.execute(sa.text(statement))
```

Comment:
What do you think about either using execute with multi=True or else using a flaky split on `;` instead of using sqlparse to avoid picking up an extra dependency (even if test only dep)